include/LinearSolver.h

```diff
@@ -25,11 +26,11 @@
 
 // Header files defining default types for template parameters.
 // These headers must be included after other MueLu/Xpetra headers.
-typedef double                                                        Scalar;
-typedef long                                                          GlobalOrdinal;
-typedef int                                                           LocalOrdinal;
-typedef Tpetra::Map<LocalOrdinal, GlobalOrdinal>::node_type           Node;
-typedef Teuchos::ScalarTraits<Scalar> STS;
+using Scalar        = Tpetra::Details::DefaultTypes::scalar_type;
```

Comment:
Can we get rid of these?

Reply:
My suspicion is that we need to keep them so that the MueLu headers below work correctly.
I could modify them as follow:
```
#include "LinearSolverTypes.h"

using Scalar        = sierra::nalu::LinSys::Scalar;
using GlobalOrdinal = sierra::nalu::LinSys::GlobalOrdinal;
using LocalOrdinal  = sierra::nalu::LinSys::LocalOrdinal;
using Node          = Tpetra::Map<LocalOrdinal, GlobalOrdinal>::node_type;
using STS           = Teuchos::ScalarTraits<Scalar>;
```
that's at least using the types defined internally by Nalu

Reply:
Done